arangod/Transaction/Methods.cpp

```diff
@@ -1717,6 +1719,16 @@ OperationResult transaction::Methods::insertLocal(std::string const& collectionN
       if (!options.isSynchronousReplicationFrom.empty()) {
         return OperationResult(TRI_ERROR_CLUSTER_SHARD_LEADER_REFUSES_REPLICATION, options);
       }
+      if (followerInfo->get()->size() + 1 < collection->minReplicationFactor()) {
+        // We cannot fulfill minimum replication Factor.
+        // Reject write.
+        LOG_TOPIC("d7306", ERR, Logger::REPLICATION)
+            << "Less then minReplicationFactor "
+            << basics::StringUtils::itoa(collection->minReplicationFactor())
+            << " followers in sync. Shard  " << collection->name()
+            << " is temporarily in read-only mode.";
+        return OperationResult(TRI_ERROR_ARANGO_READ_ONLY, options);
+      }
```

Comment:
Just to make sure, this is on a per shard basis? So we do not respect other shards that are distributed like this one?

Reply:
Would be nice if we had the actual collection name available for the error message. :roll_eyes:

Reply:
@maierlars Yes per shard basis. We did not introduce any additional changes that use distributeLike.
@goedderz yes... Operation result is UNABLE to take a proper error, and I honestly do not want to refactor too much things right now.